File: packages/platform/src/ipc/feedback.ts
/**
 * 意见反馈 IPC 封装 — 本地反馈记录 CRUD。
 *
 * 后端：壳层 `commands/feedback.rs`（InMemoryFeedbackStore + app::feedback::FeedbackService）。
 *
 * @author agent
 * @created 2026-08-13
 */

import { call } from "./invoke";

/** 反馈类型（与 Rust `FeedbackKind` 对齐）。 */
export type FeedbackKind = "feature" | "bug" | "other";

/** 反馈记录（与 Rust `Feedback` 对齐）。 */
export interface Feedback {
  id: number;
  owner_id: number;
  kind: FeedbackKind;
  title: string;
  content: string;
  created_at: string | null;
}

const OWNER_ID = 1; // 桌面单用户；多用户时由登录态注入

/** 分页查询反馈。 */
export function feedbackList(query: {
  page: number;
  page_size: number;
  kind?: FeedbackKind;
  keyword?: string;
}): Promise<[Feedback[], number]> {
  return call<[Feedback[], number]>("feedback_list", {
    request: {
      owner_id: OWNER_ID,
      page: query.page,
      page_size: query.page_size,
      kind: query.kind ?? "",
      keyword: query.keyword ?? "",
    },
  });
}

/** 新建反馈。 */
export function feedbackCreate(
  feedback: Omit<Feedback, "id" | "owner_id">,
): Promise<Feedback> {
  return call<Feedback>("feedback_create", {
    ownerId: OWNER_ID,
    feedback: { ...feedback, id: 0, owner_id: OWNER_ID },
  });
}

/** 删除反馈。 */
export function feedbackDelete(feedbackId: number): Promise<void> {
  return call<void>("feedback_delete", {
    request: { owner_id: OWNER_ID, feedback_id: feedbackId },
  });
}

File: apps/desktop/src-tauri/src/platforms/xianyu/ipc/feedback.rs
//! 意见反馈 Tauri commands。

use crate::platforms::xianyu::persist::InMemoryFeedbackStore;
use crate::shared::ipc::IpcResponse;
use app::feedback::{Feedback, FeedbackQuery, FeedbackService};
use common;
use serde::Deserialize;
use std::sync::Arc;
use tauri::State;

/// 反馈服务句柄（setup 时注册到 Tauri 状态）。
pub struct FeedbackHandle {
    pub store: Arc<InMemoryFeedbackStore>,
}

#[derive(Debug, Deserialize)]
pub struct FeedbackListRequest {
    pub owner_id: i64,
    pub page: u32,
    pub page_size: u32,
    #[serde(default)]
    pub kind: String,
    #[serde(default)]
    pub keyword: String,
}

#[derive(Debug, Deserialize)]
pub struct FeedbackDeleteRequest {
    pub owner_id: i64,
    pub feedback_id: i64,
}

#[tauri::command]
pub fn feedback_list(
    state: State<'_, FeedbackHandle>,
    request: FeedbackListRequest,
) -> common::OpenDeskResult<IpcResponse<(Vec<Feedback>, u32)>> {
    let service = FeedbackService::new(state.store.as_ref());
    let query = FeedbackQuery {
        page: request.page,
        page_size: request.page_size,
        kind: request.kind,
        keyword: request.keyword,
    };
    let result = service.list(request.owner_id, &query)?;
    Ok(IpcResponse::ok(result))
}

#[tauri::command]
pub fn feedback_create(
    state: State<'_, FeedbackHandle>,
    owner_id: i64,
    feedback: Feedback,
) -> common::OpenDeskResult<IpcResponse<Feedback>> {
    let service = FeedbackService::new(state.store.as_ref());
    let result = service
        .create(owner_id, feedback)
        .map_err(common::OpenDeskError::wrap)?;
    Ok(IpcResponse::ok(result))
}

#[tauri::command]
pub fn feedback_delete(
    state: State<'_, FeedbackHandle>,
    request: FeedbackDeleteRequest,
) -> common::OpenDeskResult<IpcResponse<()>> {
    let service = FeedbackService::new(state.store.as_ref());
    service.delete(request.owner_id, request.feedback_id)?;
    Ok(IpcResponse::ok(()))
}
